### PR TITLE
feat: angular 2 dart analysis

### PR DESCRIPTION
**Pull request by keertip**
Initial commit for Angular 2 Dart analysis support  

@yjbanov , @scheglov

### CODE REVIEW

**Comment by scheglov** [comment]
LGTM

File: modules/angular2/src/analysis/analyzer_plugin/plugin.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
+library angular2.src.analysis.analyzer_plugin;
+
+import 'package:analyzer/plugin/plugin.dart';
+
+/// Contribute a plugin to the dart analyzer for analysis of
+/// Angular2 dart code.
```
Comment (yjbanov) [inline modules/angular2/src/analysis/analyzer_plugin/plugin.dart]: nit: we refer to the new version of Angular as "Angular 2" (i.e. space before "2")

File: modules/angular2/src/analysis/server_plugin/plugin.dart [hunk 2]
```diff
@@ -0,0 +1,23 @@
+library angular2.src.analysis.server_plugin;
+
+import 'package:analyzer/plugin/plugin.dart';
+
+/// Contribute a plugin for services such as completions, indexing and refactoring
+///  of Angular2 dart code.
```
Comment (yjbanov) [inline modules/angular2/src/analysis/server_plugin/plugin.dart]: same nit as above + there's an extra space in front of "of"

File: modules/angular2/src/analysis/analyzer_plugin/plugin.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+class AngularAnalyzerPlugin extends Plugin {
```
Comment (yjbanov) [inline modules/angular2/src/analysis/analyzer_plugin/plugin.dart]: Is `Plugin` intended to be extended or implemented? From the docs it looks like an interface.

File: modules/angular2/src/analysis/server_plugin/plugin.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+class AngularServerPlugin extends Plugin {
```
Comment (yjbanov) [inline modules/angular2/src/analysis/server_plugin/plugin.dart]: Same comment as above re "extends" vs "implements"

**Comment by yjbanov** [comment]
Option: I know there isn't much in this PR, but it would still be nice to have a test that just instantiates the plugin classes. It would at least verify that the code is runnable.

**Comment by yjbanov** [comment]
Do not forget to use "git push" (*without* -f or --force)

**Comment by yjbanov** [comment]
Oops, didn't mean to close

File: modules/angular2/src/analysis/analyzer_plugin/plugin.dart (hunk 1 above)
Comment (keertip) [inline modules/angular2/src/analysis/analyzer_plugin/plugin.dart]: done.

File: modules/angular2/src/analysis/server_plugin/plugin.dart (hunk 2 above)
Comment (keertip) [inline modules/angular2/src/analysis/server_plugin/plugin.dart]: fixed

File: modules/angular2/src/analysis/analyzer_plugin/plugin.dart (hunk 3 above)
Comment (keertip) [inline modules/angular2/src/analysis/analyzer_plugin/plugin.dart]: yes, it should be implements. changed it

File: modules/angular2/src/analysis/server_plugin/plugin.dart (hunk 4 above)
Comment (keertip) [inline modules/angular2/src/analysis/server_plugin/plugin.dart]: changed

**Pull request closed without merging**